clients/src/main/java/org/apache/kafka/clients/ClientResponse.java

```diff
@@ -62,6 +67,10 @@ public boolean wasDisconnected() {
         return disconnected;
     }
 
+    public boolean versionMismatch() {
```

Comment:
This should be used any time that there is a version mismatch problem between the message we are trying to send and the message versions we can actually send to the desired broker.  I fixed the Fetcher to use this instead of doing its own thing to detect version problems.  I also change the type of this to RuntimeException, so that the exact exception which the builder raised can be preserved here.